src/db/drivers/mysql.ts

```diff
@@ -0,0 +1,219 @@
+   * @param password password used to authenticate against the database
+   * @param hostName host of the MySQL database
+   */
+  constructor(databaseName: string, username: string, password: string, host: string) {
+    super();
+
+    this.sequelize = new Sequelize(
+      databaseName,
+      username,
+      password,
+      {
+        host: host,
+        dialect: "mysql"
+      }
+    );
+
+    // define our user model for sequelize
+    this.sequelizeUserModel = this.sequelize.define("User", {
+      id: {
+        primaryKey: true,
+        type: DataTypes.INTEGER.UNSIGNED,
+        autoIncrement: true
+      },
+      username: {
+        type: new DataTypes.STRING(128),
+        allowNull: false,
+      }
+    }) as StaticSequelizeUserModel;
+    // define our study group model for sequelize
+    this.sequelizeStudyGroupModel = this.sequelize.define("StudyGroup", {
+      id: {
+        primaryKey: true,
+        type: DataTypes.INTEGER.UNSIGNED,
+        autoIncrement: true
+      },
+      groupName: {
+        type: new DataTypes.STRING(128),
+        allowNull: false,
+      }
+    }) as StaticSequelizeStudyGroupModel;
+  }
+
+  /**
+   * Confirms authentication with the database and sets up the required tables.
+   *
+   * @returns a promise that attempts to authenticate with the database and set up the tables
+   */
+  public setUp(): Promise<string> {
+    return new Promise((resolve, reject) => {
+      this.sequelize.authenticate()
+        // if we successfully authenticated with the database, set up our tables
+        .then(() => {
+          // if using a test database, drops any previously created tables
+          // TODO: look into using migration and the Sequelize CLI in production
+          this.sequelize.sync({ force: true, match: /_test$/ })
+            .then(() => resolve("Connected to MySQL database!"))
+            .catch(err => reject(`Model synchronization failed: ${err}`));
+        })
+        .catch(err => reject(`Failed to connect to MySQL database: ${err}`));
+    });
+  }
+
+  /**
+   * Gets a user from the database given their ID.
+   *
+   * @param id ID of the user to get
+   * @returns promise containing the desired user if resolved,
+   *          and an error message if rejected
+   */
+  public getUser(id: number): Promise<User> {
+    return new Promise((resolve, reject) => {
+      this.sequelizeUserModel.findByPk(id, { rejectOnEmpty: true })
+        .then(mySqlUser => {
+          const user = new User(mySqlUser.id, mySqlUser.username);
+          resolve(user);
+        })
+        .catch(err => reject(`Failed to get user: ${err}`));
+    });
+  }
+
+  /**
+   * Adds a user to the users table.
+   *
+   * @param username username of the user to create
+   * @returns promise containing newly created user if resolved,
+   *          and an error message if rejected
+   */
+  public createUser(username: string): Promise<User> {

```

Comment:
Perhaps add "TODO: add password field with proper hashing for security" to remind us that this method is incomplete